### ISSUE TITLE
MKS TFT 3.5 firmware is bad after cancel a print job 

### ISSUE THREAD

**Issue by daniellezama** [makerbase-mks/MKS-TFT#443]
Hi I need to restart the printer after cancel a job ir end one. The second job make the gcode without warming extrusion and homing. 

i am using latest firmware update. 1.0.7

Regards

Daniel Leza,a

**Comment by daniellezama** [comment]
The printer when starts at once all ok. First Job ends correctly, wifi sended. When ends, and try to send a second job or print other job from sd, start running in the air without g28 home and no heating, must turn off the printer, then come back ok. Every time must restart the printer for make work it.

**Comment by daniellezama** [comment]
i have retested today with lcd12864 display from SD card marlin runs great all ok. No problems. Then i have retested the TFT and the problem remains. Only one print job sucessful. Second job runs wihout warming without extrusion and without homing. I am sure that is a firmware bug. Comunication is fine in 57600 bauds

**Comment by daniellezama** [comment]
Well after trying the other versions of firmwares (change config settings disable babasteps) and change bauds configuration too ( i have tested in 57600 112500 an 250000) could conclude 
--> version 1.05 stable but no gcode window and no tft image when printing) ,
-->version 1.06 stable with gcode window and tft image when printing, but when send the program remotely not show the print image. 
-->version  1.07 unstable show the print image and run fine only first printing. next prints in the air. Also problems for detect the sd ( must shut down and restart for read the SD).

Finally my choice version 1.0.6 less problems I have.

Waiting for version 1.08 (Hope MKS finally fix the problems)

Regards

**Comment by jechs83** [comment]
Same problem here with 1.07, it looks  really nice with more configurations functions but its impossible to print normally after first print

**Comment by CBDesignS** [comment]
yes I have the same with 1.0.7 as above. it seems to just continue from where it was stopped without re leveling or even heating or extrusion. . I think tft pretends to be a computer controlling main board by serial data but the serial data is not reset so the tft just starts from the last command without realising it has been stopped, cooled down etc. bad firmware coding or config. please sort this out asap. 

tft35 connected to mks sgen_l v1 (clean build of marlin 2.0.xx bugfix from mid aug) also with 12864 reprap lcd attached.

**Comment by daniellezama** [comment]
Yes i have the same configuration CBDesignS and must choose the 1.0.6 version more stable and works. Only problem is when send the tft gcode from computer  not show the tft image when print. 

Regards

Daniel

**Comment by CBDesignS** [comment]
daniellezama I had a simmilar problem with the tft gcode preview messing up untill I updated cura to 4.7.0 and manually updated the tft/wifi plugin from the github code page and now the preview is ok.
![20200901_225346](https://user-images.githubusercontent.com/54717994/91911737-a16b1880-eca9-11ea-902d-5a7df393485e.jpg)

before it was random lines and only part of the preview

**Comment by jechs83** [comment]
Still waiting tft 35 firmware 1.0.8 and makerbase doesnt care about it...  I think we should try other brand options by now.

**Comment by CBDesignS** [comment]
I removed the mks hardware and jumped to btt because at least the userbase  is willing to support the hardware better than bigtree is doing and MKS ever will.  The TFT has just as many problems but at least users are skilled enough to fix the problems that have arrise from time to time
